### PR TITLE
docs(mesh) Add docs for mesh RBAC for new policies

### PR DESCRIPTION
**Pull request by Automaat**
Signed-off-by: Marcin Skalski <marcin.skalski@konghq.com>

### Summary
Extend Kong Mesh docs with RBAC to include usage of new policies introduces in 2.0 version

### Reason
https://github.com/Kong/kong-mesh/issues/1874

### CODE REVIEW

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:90]:
```suggestion
    mesh: default # Grants access to the resources in the named mesh. It can only be used with the mesh-scoped resources.
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:91]:
```suggestion
    access: ["CREATE", "UPDATE", "DELETE"] # The action bound to a type.
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:92]:
```suggestion
    when: # A set of qualifiers to receive access. Only one of them needs to be fulfilled to receive access.
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:93]:
```suggestion
    - tagetRef: # A condition on the targetRef section in policies 2.0 (like MeshAccessLog or MeshTrace).
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:108]: I think all the suggestions I had in the last tab above would apply to this tab as well.

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:282]:
```suggestion
> **Note**: When giving users `UPDATE` permission, remember to add `UPDATE` permission to all selectors they can switch between. For example, if a user only has access to `sources` selector, they won't be able to update policy with `destinations` selector or new `targetRef` selectors. Likewise, when a user only has access to `targetRef` kind `MeshService`, they won't be able to update the policy to use a different `targetRef` kind.
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:759]:
```suggestion
    mkdir -p /tmp/k8s-certs
    cd /tmp/k8s-certs
    openssl genrsa -out backend-owner.key 2048 # generate client key
    openssl req -new -key backend-owner.key -subj "/CN=backend-owner" -out backend-owner.csr # generate client certificate request
    CSR=$(cat backend-owner.csr | base64 | tr -d "\n") && echo "apiVersion: certificates.k8s.io/v1
    kind: CertificateSigningRequest
    metadata:
      name: backend-owner
    spec:
      request: $CSR
      signerName: kubernetes.io/kube-apiserver-client
      usages:
      - client auth" | kubectl apply -f -
    kubectl certificate approve backend-owner
    kubectl get csr backend-owner -o jsonpath='{.status.certificate}'| base64 -d > backend-owner.crt
    kubectl config set-credentials backend-owner \
    --client-key=/tmp/k8s-certs/backend-owner.key \
    --client-certificate=/tmp/k8s-certs/backend-owner.crt \
    --embed-certs=true
    kubectl config set-context backend-owner --cluster=YOUR_CLUSTER_NAME --user=backend-owner
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:765]:
```suggestion
    echo "
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:803]:
```suggestion
    echo "
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:824]:
```suggestion
    echo '
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:857]:
```suggestion
    kubectl config use-context backend-owner
    echo "
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:878]:
```suggestion
    echo "
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:903]:
```suggestion
> **Note**: By default, all requests that originate from localhost are authenticated as the `admin` user in the `mesh-system:admin` group.
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:904]:
```suggestion
For this example to work, you must either run the control plane with `KUMA_API_SERVER_AUTHN_LOCALHOST_IS_ADMIN` set to `false` or access the control plane using a method other than localhost.
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:910]:
```suggestion
    export ADMIN_TOKEN=$(curl http://localhost:5681/global-secrets/admin-user-token | jq -r .data | base64 -d)
    kumactl config control-planes add \
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:928]:
```suggestion
    export BACKEND_OWNER_TOKEN=$(kumactl generate user-token --valid-for=24h --name backend-owner)
    kumactl config control-planes add \
    --name=cp-backend-owner \
    --address=https://localhost:5682 \
    --skip-verify=true \
    --auth-type=tokens \
    --auth-conf token=$BACKEND_OWNER_TOKEN
    kumactl config control-planes switch --name cp-admin # switch back to admin
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:934]:
```suggestion
    echo "type: AccessRoleBinding
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:943]:
```suggestion
1.  Create {{site.mesh_product_name}} RBAC to only allow the backend-owner to modify `TrafficPermission` for backend:
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:958]:
```suggestion
    echo '
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:972]:
```suggestion
    kumactl config control-planes switch --name cp-backend-owner
    echo "
```

File: src/mesh/features/rbac.md
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:989]:
```suggestion
    echo "
```

**Review by cloudjumpercat — CHANGES_REQUESTED** [review]
@lahabana I left a few copyedits to clarify some of the wording and removed "$" from the code examples.

**Review by lahabana — APPROVED** [review]
LGTM once @cloudjumpercat 's suggestions are included

File: src/mesh/features/rbac.md
Comment (Automaat) [reply src/mesh/features/rbac.md]: Updated second tab

**Review by cloudjumpercat — APPROVED** [review]

**Comment by cloudjumpercat** [comment]
@Automaat I think if you rebase this against `release/mesh-2.0`, it will fix the broken link checks. I just merged the updated nav in that branch, so that should resolve the link problems.

**Comment by Automaat** [comment]
something bad happend after merge, reopened this pr here: https://github.com/Kong/docs.konghq.com/pull/4699

**Pull request closed without merging**
